staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation_test.go

```diff
@@ -1683,17 +1598,34 @@ func TestValidationExpressions(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			// set costBuget to maxInt64 for current test
+			tt.costBudget = math.MaxInt64
 			for _, validRule := range tt.valid {
 				t.Run(validRule, func(t *testing.T) {
 					s := withRule(*tt.schema, validRule)
 					celValidator := NewValidator(&s)
 					if celValidator == nil {
 						t.Fatal("expected non nil validator")
 					}
-					errs := celValidator.Validate(field.NewPath("root"), &s, tt.obj)
+					errs, _ := celValidator.Validate(field.NewPath("root"), &s, tt.obj, tt.costBudget)
 					for _, err := range errs {
 						t.Errorf("unexpected error: %v", err)
 					}
+
+					// test with cost budge exceed
+					errs, _ = celValidator.Validate(field.NewPath("root"), &s, tt.obj, 0)
+					var found bool
+					for _, err := range errs {
+						if err.Type == field.ErrorTypeInvalid || strings.Contains(err.Error(), "validation failed due to cost budget exceed") {
```

Comment:
why the general `||` against `ErrorTypeInvalid`? do the cost errors for valid rules not consistently come back in an identifiable way?

Reply:
Fixed with `&&`